sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/HttpHealthcareApiClient.java

```diff
@@ -545,6 +529,26 @@ public HttpBody readFhirResource(String resourceId) throws IOException {
     return client.projects().locations().datasets().fhirStores().fhir().read(resourceId).execute();
   }
 
+  @Override
+  public HttpBody searchFhirResource(
+          String fhirStore,
+          String resourceType,
+          @Nullable Map<String, Object> parameters)
+          throws IOException {
+    SearchResourcesRequest request = new SearchResourcesRequest().setResourceType(resourceType);
+    Search search = client
+            .projects()
+            .locations()
+            .datasets()
+            .fhirStores()
+            .fhir()
+            .search(fhirStore, request);
```

Comment:
You will need to handle pagination here, see how this is handled for ListMessages: https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/HttpHealthcareApiClient.java#L682.